hawkular-inventory-impl-tinkerpop-parent/hawkular-inventory-impl-tinkerpop/src/main/java/org/hawkular/inventory/impl/tinkerpop/FilterVisitor.java

```diff
@@ -85,7 +90,7 @@ public void visit(GraphTraversal<?, ?> query, Related related, QueryTranslationS
                 if (null != related.getRelationshipName()) {
                     state.setInEdges(true);
                     state.setComingFrom(Direction.IN);
-                    query.inE(related.getRelationshipName());
+                    query.inE(related.getRelationshipName()).discriminate(discriminator);
```

Comment:
Well I didn't mean to do something really global, but just in the scope of the "query" object. What I don't know is, if you write "query.discriminate(disc).inE(related.getRelationshipName())" => is the discriminator propagated? In other words, is it equivalent to "query.inE(related.getRelationshipName()).discriminate(disc)" ?